centreon/src/Core/Dashboard/Playlist/Application/UseCase/CreatePlaylist/CreatePlaylist.php

```diff
@@ -0,0 +1,221 @@
+
+        return $response;
+    }
+
+    /**
+     * Validate that name is unique and dashboards exists.
+     *
+     * @param CreatePlaylistRequest $request
+     *
+     * @throws PlaylistException|\Throwable
+     */
+    private function validateNameAndDashboards(CreatePlaylistRequest $request): void
+    {
+        $this->info(
+            'Validating playlist name and dashboards',
+            [
+                'playlist_name' => $request->name,
+                'dashboards' => $request->dashboards,
+            ]
+        );
+        $this->validator->validatePlaylistNameIsUnique($request->name);
+        if ([] !== $request->dashboards) {
+            $dashboardIds = array_map(static fn (array $dashboard): int => $dashboard['id'], $request->dashboards);
+            $this->validator->validateDashboardExists($dashboardIds);
+            $this->validator->validateDashboardIsUnique($dashboardIds);
+            if (! $this->rights->hasAdminRole()) {
+                $this->validator->validateUserHasAccessToDashboards($dashboardIds, $this->user);
+            }
+        }
+    }
+
+    /**
+     * Create NewPlaylist Entity.
+     *
+     * @param CreatePlaylistRequest $request
+     *
+     * @throws \Assert\AssertionFailedException
+     * @throws PlaylistException
+     *
+     * @return NewPlaylist
+     */
+    private function createNewPlaylistModel(CreatePlaylistRequest $request): NewPlaylist
+    {
+        $newPlaylist = (new NewPlaylist($request->name, $request->rotationTime, $request->isPublic))->setAuthor(
+            new PlaylistAuthor($this->user->getId(), $this->user->getAlias())
+        );
+        if ($request->description !== null) {
+            $newPlaylist->setDescription($request->description);
+        }
+        if ([] !== $request->dashboards) {
+            $dashboardsOrder = [];
+            foreach ($request->dashboards as $dashboard) {
+                $dashboardsOrder[] = new DashboardOrder($dashboard['id'], $dashboard['order']);
+            }
+            $newPlaylist->setDashboardsOrder($dashboardsOrder);
+        }
+
+        return $newPlaylist;
+    }
+
+    /**
+     * @param NewPlaylist $newPlaylist
+     *
+     * @throws \Throwable
+     *
+     * @return int
+     */
+    private function writePlaylist(NewPlaylist $newPlaylist): int
+    {
+        try {
+            $this->dataStorageEngine->startTransaction();
+            $this->info('add playlist in data storage');
+            $playlistId = $this->writePlaylistRepository->add($newPlaylist);
+            $this->info('add dashboards <=> playlist relation in data storage');
+            $this->writePlaylistRepository->addDashboardsToPlaylist($playlistId, $newPlaylist->getDashboardsOrder());
+            $this->dataStorageEngine->commitTransaction();
+        } catch (\Throwable $ex) {
+            $this->error('An error occured, transaction rollback');
+            $this->dataStorageEngine->rollbackTransaction();
```

Comment:
We don't have any call to `isAlreadyinTransaction()`, is that normal :thinking: ?